From 2a4fed8331f996421e65db446559991a854e2ad3 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 24 Mar 2023 18:23:01 -0700
Subject: [PATCH] m4: Check for FTW_ACTIONRETVAL along with nftw

FTW_ACTIONRETVAL is glibc specific extention which is used to implement
xfsfind but it may not be available on other C library implementations on Linux
e.g. musl. Therefore ensure that these defines are available before declaring
nftw() to be usable

Upstream-Status: Submitted [https://lore.kernel.org/fstests/20230325012858.587801-1-raj.khem@gmail.com/T/#u]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
Cc: Darrick J. Wong <djwong@kernel.org>
Cc: Zorro Lang <zlang@redhat.com>
---
 m4/package_libcdev.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/m4/package_libcdev.m4 b/m4/package_libcdev.m4
index b41c087b..7f731044 100644
--- a/m4/package_libcdev.m4
+++ b/m4/package_libcdev.m4
@@ -132,7 +132,7 @@ AC_DEFUN([AC_HAVE_NFTW],
 #include <stddef.h>
 #include <ftw.h>
     ]], [[
-         nftw("/", (int (*)(const char *, const struct stat *, int, struct FTW *))1, 0, 0);
+         nftw("/", (int (*)(const char *, const struct stat *, int, struct FTW *))1, 0, FTW_ACTIONRETVAL);
     ]])],[have_nftw=yes
        AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)])
     AC_SUBST(have_nftw)
-- 
2.40.0

